execution-engine/contracts/examples/simple-token/src/lib.rs

```diff
@@ -0,0 +1,117 @@
+
+    // For stored contract
+    let mut stored_contract_urefs: BTreeMap<String, Key> = BTreeMap::new();
+    stored_contract_urefs.insert(String::from(TOKEN_MAP_NAME), token_map_turef.into());
+    stored_contract_urefs.insert(String::from(TOTAL_SUPPLY_NAME), token_supply_turef.into());
+}
+
+fn mint(address: Key, value: U512) -> Option<bool>{
+    if runtime::has_key(TOKEN_MAP_NAME) {
+        let token_map_uref: TURef<BTreeMap<Key, U512>> = runtime::get_key(TOKEN_MAP_NAME)
+            .unwrap_or_revert_with(ApiError::GetKey)
+            .try_into().unwrap_or_revert();
+        let mut token_map = storage::read(token_map_uref.clone())
+            .unwrap_or_revert_with(ApiError::Read)
+            .unwrap_or_revert_with(ApiError::ValueNotFound);
+
+        // If token_map has given address, update
+        // Or, insert value
+        let val = token_map.entry(address).or_insert(U512::from(0));
+        *val += value;
+
+        storage::write(token_map_uref, token_map);
+    }
+
+    if runtime::has_key(TOTAL_SUPPLY_NAME) {
+        let total_supply_uref: TURef<U512> = runtime::get_key(TOTAL_SUPPLY_NAME)
+            .unwrap_or_revert()
+            .try_into().unwrap_or_revert();
+        let total_supply_opt = storage::read(total_supply_uref)
+            .unwrap_or_revert();
+        let total_supply_after = match total_supply_opt {
+            Some(total_supply_before) => {
+                total_supply_before + value
+            }
+            _ => panic!("Total supply calc error!"),
+        };
+        storage::write(total_supply_uref, total_supply_after);
+    }
+
+    Some(true)
+}
+
+fn get_total_supply() -> Option<U512>{
+    let total_supply_uref: TURef<U512> = runtime::get_key(TOTAL_SUPPLY_NAME)
+        .unwrap_or_revert_with(ApiError::GetKey)
+        .try_into().unwrap_or_revert();
+    let total_supply = storage::read(total_supply_uref)
+        .unwrap_or_revert();
+    
+    total_supply
+}
+
+#[no_mangle]
+pub extern "C" fn call() {
+    let method_name: String = runtime::get_arg(0)
+        .unwrap_or_revert_with(ApiError::MissingArgument)
+        .unwrap_or_revert_with(ApiError::InvalidArgument);
+
+    match method_name.as_str() {
+        "init" => init_token_db(),
+        "mint" => {
+            let address: Key = runtime::get_arg(1)
+                .unwrap_or_revert_with(ApiError::MissingArgument)
+                .unwrap_or_revert_with(ApiError::MissingArgument);
+            let amount: U512 = runtime::get_arg(2)
+                .unwrap_or_revert_with(ApiError::MissingArgument)
+                .unwrap_or_revert_with(ApiError::MissingArgument);
+            let result: Option<bool> = mint(address, amount);
+            match result {
+                Some(true) => {
+                    let ret = CLValue::from_t(true).unwrap_or_revert();
+                    runtime::ret(ret)
+                }
+
+                _ => panic!("Error in init step!"),
```

Comment:
I recommend to use `unwrap_or_revert`, `runtime::revert` in `contracts` instead of `panic!` to get information about the failure.